uai2017/python/populate_papers_data.py

```diff
@@ -0,0 +1,152 @@
+import os
+import sys
+import json
+import argparse
+from openreview import *
+sys.path.append(os.path.join(os.getcwd(), "../../dto/uai2017"))
+print sys.path
+from note_content import *
+from constants import *
+from lxml import etree
+
+"""
+Parse the arguments for user authentication
+"""
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base URL")
+parser.add_argument('--overwrite', help="If set to true, overwrites existing groups")
+parser.add_argument('--username')
+parser.add_argument('--password')
+
+args = parser.parse_args()
+if args.username != None and args.password != None:
+    openreview = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+else:
+    openreview = Client(baseurl=args.baseurl)
+
+
+def get_paper_names(notes):
+    """
+    Getting paper name using the following logic: If note.number is NONE then use 0 as default else 'Paper' followed by its number
+    :param notes:
+    :return:
+    """
+    list_paper_name = []
+    for note in notes:
+        if (note.number is None):
+            list_paper_name.append("Paper0")
+        else:
+            list_paper_name.append("Paper" + str(note.number))
+    return list_paper_name
+
+
+def get_notes_submitted_papers():
+    """
+    Get all the submitted papers
+    :return:
+    """
+    notes = openreview.get_notes(invitation=CONFERENCE_SUBMISSION)
+    return notes
+
+
+def get_paper_metadata_notes(paper_names):
+    """
+    Get all submitted paper meta data notes
+    :return:
+    """
+    notes = [openreview.get_notes(invitation=CONFERENCE + "/" + paper_name + "/-/matching")[0] for paper_name in
+             paper_names]
+    return notes
+
+
+def get_submitted_paper_details(submitted_paper_notes):
+    """
+    Get details of all the papers submitted in a list
+    :param submitted_paper_notes:
+    :return:
+    """
+    paper_details_list=[]
+    paper_details_list.append(["Paper ID","Paper Title","Abstract","Author Names","Author Emails","Subject Areas","Conflict Reasons",
+                               "Files","Supplementary File","Related Submissions","AC suggestions","Toronto Paper Matching System Agreement"])
+    for note in submitted_paper_notes:
+        paper_id = note.id
+        paper_title = note.content["title"]
+        paper_abstract = note.content["abstract"]
+        author_names= note.content["authors"]
+        author_emails = note.content["author_emails"]
```

Comment:
We renamed this field to "authorids" and "authors", "authorids" and "conflicts" are arrays instead of strings. See the ICLR2017 conference submission invitation